### DISCUSSION TITLE
Version 2.9.0

### DISCUSSION THREAD

**Discussion by henryiii** [pybind/pybind11#3576] [Announcements]
This is the last version to support Python 2.7 and 3.5.

New Features:

-   Allow `py::args` to be followed by other arguments; the remaining arguments are implicitly keyword-only, as if a `py::kw_only{}` annotation had been used. [#3402](https://github.com/pybind/pybind11/pull/3402)

Changes:

-   Make str/bytes/memoryview more interoperable with `std::string_view`. [#3521](https://github.com/pybind/pybind11/pull/3521)
-   Replace `_` with `const_name` in internals, avoid defining `pybind::_` if `_` defined as macro (common gettext usage) [#3423](https://github.com/pybind/pybind11/pull/3423)

Bug fixes:

-   Fix a rare warning about extra copy in an Eigen constructor. [#3486](https://github.com/pybind/pybind11/pull/3486)
-   Fix caching of the C++ overrides. [#3465](https://github.com/pybind/pybind11/pull/3465)
-   Add missing `std::forward` calls to some `cpp_function` overloads. [#3443](https://github.com/pybind/pybind11/pull/3443)
-   Support PyPy 7.3.7 and the PyPy3.8 beta. Test python-3.11 on PRs with the `python dev` label. [#3419](https://github.com/pybind/pybind11/pull/3419)
-   Replace usage of deprecated `Eigen::MappedSparseMatrix` with `Eigen::Map<Eigen::SparseMatrix<...>>` for Eigen 3.3+. [#3499](https://github.com/pybind/pybind11/pull/3499)
-   Tweaks to support Microsoft Visual Studio 2022. [#3497](https://github.com/pybind/pybind11/pull/3497)

Build system improvements:

-   Nicer CMake printout and IDE organisation for pybind11's own tests. [#3479](https://github.com/pybind/pybind11/pull/3479)
-   CMake: report version type as part of the version string to avoid a spurious space in the package status message. [#3472](https://github.com/pybind/pybind11/pull/3472)
-   Flags starting with `-g` in `$CFLAGS` and `$CPPFLAGS` are no longer overridden by `.Pybind11Extension`. [#3436](https://github.com/pybind/pybind11/pull/3436)
-   Ensure ThreadPool is closed in `setup_helpers`. [#3548](https://github.com/pybind/pybind11/pull/3548)
-   Avoid LTS on `mips64` and `ppc64le` (reported broken). [#3557](https://github.com/pybind/pybind11/pull/3557)

<hr /><em>This discussion was created from the release <a href='https://github.com/pybind/pybind11/releases/tag/v2.9.0'>Version 2.9.0</a>.</em>

**Comment by Trigve** [comment]
Congratulations for the new release!

Question (@rwgk): When will be smart_holder branch merged to the "master" branch? I was hoping 2.9.0 would contain the smart_holder branch stuff.

**Reply by rwgk** [reply]
I'm finally getting back to it.
Big picture: I don't expect a lot of changes relative to what's currently on the smart_holder branch.
Current thinking:
* I want to bring PR #3581 to a conclusion (not sure exactly yet what that will be).
* tests/CMakeLists.txt needs work, to better deal with test_class_sh_module_local (help highly appreciated; my cmake skills are highly underdeveloped).
* remove non-production code from smart_holder_poc.h (EDIT 2022-01-04)
* the documentation needs a pass.
* pass reviews.

That's all I can think of at the moment.

**Reply by Trigve** [reply]
Thank you for your detailed response.
Unfortunately, my CMAKE skills are close to zero. But if you have some other areas you would appreciate any help in, I could take look at it (even with my limited spare time).
